lib/ex_sdp/attribute.ex

```diff
@@ -12,116 +12,56 @@ defmodule ExSDP.Attribute do
-
-  @type t :: %__MODULE__{key: key(), value: value()}
-
-  @flag_values @flag_attributes |> Enum.map(&to_string/1)
-  @directly_assignable_values @value_attributes |> Enum.map(&to_string/1)
-  @numeric_values @numeric_attributes |> Enum.map(&to_string/1)
+  @type t :: __MODULE__.RTPMapping.t() | {key(), value()} | value()
 
   @doc """
-  Parses SDP Attribute.
+  Parses SDP Attribute line.
 
-  `t:value_attributes/0` and `t:numeric_attributes/0` formatted as `name:value`
-  are be parsed as `{name, value}` other values are treated as
-  `t:flag_attributes/0`. Known attribute names are converted into atoms.
-  """
-  @spec parse(binary()) :: {:ok, t} | {:error, atom()}
-  def parse(line) do
-    line
-    |> String.split(":", parts: 2)
-    |> handle_known_attribute()
-  end
+  `line` is a string in form of `a=attribute` or `a=attribute:value`.
+  `opts` is a keyword list that can contain some information for parsers.
 
-  @spec parse_media_attribute(t(), atom() | binary()) :: {:error, :invalid_attribute} | {:ok, t()}
-  def parse_media_attribute(%__MODULE__{key: :rtpmap, value: value}, media) do
-    with {:ok, %RTPMapping{} = mapping} <- RTPMapping.parse(value, media) do
-      {:ok, %__MODULE__{key: :rtpmap, value: mapping}}
-    end
+  Unknown attributes keys are returned as strings, known ones as atoms.
+  """
+  @spec parse(binary(), opts :: []) :: {:ok, t()} | {:error, atom()}
+  def parse(line, opts \\ []) do
+    [attribute | value] = String.split(line, ":", parts: 2)
+    attribute = String.to_atom(attribute)
+    do_parse(attribute, List.first(value), opts)
   end
 
-  @spec parse_media_attribute(t(), atom() | binary()) :: {:error, :invalid_attribute} | {:ok, t()}
-  def parse_media_attribute(other, _), do: {:ok, other}
+  defp do_parse(flag, nil, _opts) when flag in @flag_attributes, do: {:ok, flag}
 
-  defp handle_known_attribute(["framerate", framerate]) do
-    with {:ok, framerate} <- parse_framerate(framerate) do
-      {:ok, %__MODULE__{key: :framerate, value: framerate}}
-    end
-  end
+  defp do_parse(flag, nil, _opts), do: {:ok, "#{flag}"}
 
-  defp handle_known_attribute([flag]) when is_binary(flag) and flag in @flag_values do
-    {:ok, %__MODULE__{value: String.to_atom(flag)}}
-  end
+  defp do_parse(:rtpmap, value, opts), do: RTPMapping.parse(value, opts)
 
-  defp handle_known_attribute([prop, value])
-       when is_binary(prop) and prop in @directly_assignable_values do
-    {:ok, %__MODULE__{key: String.to_atom(prop), value: value}}
+  defp do_parse(:framerate, value, _opts) do
+    case String.split(value, "/") do
+      [framerate] -> {:ok, {:framerate, String.to_float(framerate)}}
+      [left, right] -> {:ok, {:framerate, {String.to_integer(left), String.to_integer(right)}}}
+    end
   end
 
-  defp handle_known_attribute([prop, value]) when is_binary(prop) and prop in @numeric_values do
+  defp do_parse(attribute, value, _opts) when attribute in [:maxptime, :ptime, :quality] do
```

Comment:
do we specify somewhere that these attributes are known and values converted to integers?